lib/couchdb.js

```diff
@@ -497,6 +497,10 @@ CouchDB.prototype.find =
     mo.db.get(id, function(err, doc) {
       if (err && err.statusCode === 404) return cb(null, []);
       if (err) return cb(err);
+      // Check if the fetched record belong to the query model.
+      const modelView = mo.modelView;
+      const errWrongModel = new Error(`Record with id ${id} does not belong to model ${model}.`);
+      if (doc[modelView] != model) return cb(errWrongModel);
```

Comment:
I think we should also set `statusCode` to 404 to not found and add `code` to allow callers detect this error programatically.

To avoid leaking information that could be used by an attacker, I think it would be best to return a 404 error with the same message as is reported by `mo.db.get` or juggler and provide the detailed information via the debug log only.

IIUC, connectors are supposed to return an empty/falsy result from the `find` method when the target id does not exist - see L498 above or [`find` implementation in memory connector](https://github.com/strongloop/loopback-datasource-juggler/blob/a0de6bc760711736858ab9177af3196e211e2c0f/lib/connectors/memory.js#L379). I am proposing to leverage the same mechanism for this edge case too.

```suggestion
      const actualModel = doc[mo.modelView];
      if (model != actualModel) {
        debug('Rejecting request to get model %s with id %s, because the object belongs to a different model: %s',
          model, id, actualModel);
        return cb(null, []);
      }
```

Reply:
@bajtos thank you I return empty result instead and added the details in the debug information.